common/src/main/scala/tracking/BatchingTopicSubscriptionsRepository.scala

```diff
@@ -0,0 +1,94 @@
+package tracking
+import akka.actor.ActorSystem
+import models.Topic
+import tracking.Repository.RepositoryResult
+
+import scala.concurrent.Future
+import scala.concurrent.duration._
+import akka.actor.{Actor, Props}
+import akka.pattern.ask
+import akka.util.Timeout
+import cats.implicits._
+
+import scala.concurrent.duration.FiniteDuration
+
+object BatchingTopicSubscriptionsRepository {
+
+  case class SubscriptionEvent(topic: Option[Topic], topicId: String, count: Int)
+
+  case object Flush
+
+  private class InternalActor(underlying: TopicSubscriptionsRepository) extends Actor {
+
+    implicit val ec = context.dispatcher
+
+    var events = collection.mutable.Buffer[SubscriptionEvent]()
+
+    override def receive: Actor.Receive = {
+      case event: SubscriptionEvent =>
+        events.append(event)
+
+      case Flush =>
+        if (events.nonEmpty) {
+          val requestor = sender
+          flush() foreach { _ => requestor ! () }
```

Comment:
`flush()` returns a future, so I am avoiding getting the wrong value for `sender` since the actor might be processing a new message (and therefore have a new `sender`) by the time the future has completed.